### PR TITLE
Add a guide of plugging in CA cert to istiod and update CA documents

### PR DESCRIPTION
**Pull request by lei-tang**
Please provide a description for what this PR is for: https://github.com/istio/istio/issues/21696.
This PR adds a guide of plugging in CA cert to istiod and updates CA documents for Istio 1.5.
- Add a guide of plugging in a CA cert to istiod.
- Update CA documents for the istiod related changes in Istio 1.5.

[ ] Configuration Infrastructure
[ X] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/security/authorization/authz-td-migration/index.md [hunk 1]
```diff
@@ -112,7 +112,7 @@ Notice that it may take tens of seconds for the authorization policy to be propa
     Istio mesh is now running with a new trust domain, `new-td`.
 
 1. Delete secrets of `sleep` and `httpbin` in `default` namespace and in `sleep-allow` namespace. Notice if you install Istio with SDS,
```
Comment (diemtvu) [inline content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: It's probably good to explain what does (install Istio with SDS) means. You may need to differentiate the case for step 1 (i.e should users use different profile or flags to change the TD value).

File: content/en/docs/tasks/security/authorization/authz-td-migration/index.md [hunk 2]
```diff
@@ -112,7 +112,7 @@ Notice that it may take tens of seconds for the authorization policy to be propa
     Istio mesh is now running with a new trust domain, `new-td`.
```
Comment (diemtvu) [inline content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: Should we state the assumption of the prior settings. In the other words, set controlPlaneSecurityEnabled and mtls.enabled is not part of the change (though they should be on to make sense). 

We also don't want to mentioned both of these flags, as likely they will be removed in 1.6. User should set authn policy for mtls.enabled instead.

File: content/en/docs/setup/additional-setup/config-profiles/index.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ your specific needs. The following built-in configuration profiles are currently
 
 1. **minimal**: the minimal set of components necessary to use Istio's [traffic management](/docs/tasks/traffic-management/) features.
 
-1. **sds**: similar to the **default** profile, but also enables Istio's [SDS (secret discovery service)](/docs/tasks/security/citadel-config/auth-sds).
+1. **sds**: similar to the **default** profile, but also enables Istio's SDS (secret discovery service).
```
[the remark this answers is not in the document]
Comment (JimmyCYJ) [reply content/en/docs/setup/additional-setup/config-profiles/index.md]: I don't see SDS profile when testing 1.5 beta5. This profile is removed.

[the remark this answers is not in the document]
Comment (diemtvu) [reply content/en/docs/setup/additional-setup/config-profiles/index.md]: Please remove this line then. Also make sure remove all places referring to file mount cert (can be in different PR)

[the remark this answers is not in the document]
Comment (lei-tang) [reply content/en/docs/setup/additional-setup/config-profiles/index.md]: Done: this line is removed.

File: content/en/docs/tasks/security/authorization/authz-td-migration/index.md (hunk 1 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: This part is removed as no mounted secrets in the new release.

File: content/en/docs/tasks/security/authorization/authz-td-migration/index.md (hunk 2 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: As this PR focuses on Citadel related changes, this change on the flags of controlPlaneSecurityEnabled and mtls.enabled can be in a separate PR.

Comment (diemtvu) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: I think we should remove in the same CL to avoid create any side effect. The whole command can be simplify to a oneliner with --ser

Comment (lei-tang) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: A simplification sounds good to me. Can you suggest the oneliner to use?

Comment (diemtvu) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: I think this should work

`istioctl manifest apply --set values.global.trustDomain=new-td`

You can mention this assumes user previously install Istio with default profile.

Comment (lei-tang) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: Simplified to "istioctl manifest apply --set profile=demo --set values.global.trustDomain=new-td". The profile used in the guide is demo, hence the --set profile=demo.

Comment (diemtvu) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: It's ok to use profile `demo` if you want, but you should also change step 2  to the similar command.

Comment (diemtvu) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: And also in the section below (with alias)

Comment (lei-tang) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: The step 2 has been changed to use a similar command. As to the following trustDomainAliases, I am not sure about the equivalent command line since it has an array in it. In addition, td-installation.yaml is also referred in the cleanup section.  
    apiVersion: install.istio.io/v1alpha1
    kind: IstioOperator
    spec:
      values:
        global:
          controlPlaneSecurityEnabled: false
          mtls:
            enabled: true
          trustDomain: new-td
          trustDomainAliases:
            - old-td

Comment (diemtvu) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: Hmm. I don't know how to set array params neither. Keep the yaml is fine for now, but please remove mtls and controlPlaneSecurity to be consistent with other.

Comment (lei-tang) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: mtls and controlPlaneSecurity have been removed.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 4]
```diff
@@ -7,16 +7,13 @@ aliases:
     - /docs/tasks/security/plugin-ca-cert/
 ---
 
+In Istio 1.5, Citadel is linked with Pilot.
```
Comment (linsun) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: can this be reworded?   Should it be something like Istio CA from citadel is merged into istiod?

**Comment by linsun** [comment]
LG thank you @lei-tang! just a small comment to clarify things.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 4 above)
Comment (costinm) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: +1 - I don't think users whould see 'Citadel' or 'Pilot' in any new docs.

File: content/en/docs/tasks/security/ca-config/_index.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (frankbu) [inline content/en/docs/tasks/security/ca-config/_index.md]: Is this directory still needed now, since there is only 1 task under it? Should we move the task up on level instead?

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 4 above)
Comment (frankbu) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Why is this needed at all? It's an implementation detail. How about just remove this line and change the following line to:
```
This task shows how operators can configure the Istio certificate authority with an existing root certificate, signing certificate and key.
```
You also should change all occurrences of `Citadel` to `Istio's CA` in the rest of the doc.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 6]
```diff
@@ -7,16 +7,13 @@ aliases:
...
 
 By default, Citadel generates self-signed root certificate and key, and uses them to sign the workload certificates.
 Citadel can also use the operator-specified certificate and key to sign workload certificates, with
 operator-specified root certificate. This task demonstrates an example to plug certificates and key into Citadel.
 
-## Before you begin
-
-Follow the [Istio installation guide](/docs/setup/install/istioctl/) to install Istio with mutual TLS enabled.
-
 ## Plugging in the existing certificate and key
```
Comment (frankbu) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]:
```suggestion
## Plugging in an existing certificate and key
```

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 7]
```diff
@@ -7,16 +7,13 @@ aliases:
...
 
 Suppose we want to have Citadel use the existing signing (CA) certificate `ca-cert.pem` and key `ca-key.pem`.
```
Comment (frankbu) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]:
```suggestion
Suppose we want to have Citadel use an existing signing (CA) certificate `ca-cert.pem` and key `ca-key.pem`.
```

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 8]
```diff
@@ -38,66 +35,53 @@ These files are ready to use in the `samples/certs/` directory.
...
 
     {{< text bash >}}
+    $ kubectl create namespace istio-system
     $ kubectl create secret generic cacerts -n istio-system --from-file=samples/certs/ca-cert.pem \
         --from-file=samples/certs/ca-key.pem --from-file=samples/certs/root-cert.pem \
         --from-file=samples/certs/cert-chain.pem
     {{< /text >}}
 
-1.  Redeploy Citadel with `global.mtls.enabled` set to `true` and `security.selfSigned` to `false`.
+1.  Deploy Citadel with `global.mtls.enabled` set to `true` and the `demo` profile.
```
Comment (frankbu) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]:
```suggestion
1.  Deploy Istio using the `demo` profile and with `global.mtls.enabled` set to `true`.
```

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 9]
```diff
@@ -38,66 +35,53 @@ These files are ready to use in the `samples/certs/` directory.
...
-    The file `/tmp/pod-root-cert.pem` contains the root certificate propagated to the pod.
+1.  Retrieve the certificate chain of `httpbin`.
 
     {{< text bash >}}
-    $ kubectl exec -it $RATINGSPOD -c istio-proxy -- /bin/cat /etc/certs/cert-chain.pem > /tmp/pod-cert-chain.pem
+    $ kubectl exec $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name}) -c istio-proxy -n foo -- openssl s_client -showcerts -connect httpbin.foo:8000 > httpbin-proxy-cert.txt
     {{< /text >}}
 
-    The file `/tmp/pod-cert-chain.pem` contains the workload certificate and the CA certificate propagated to the pod.
+    Open `httpbin-proxy-cert.txt` created in the above command and save the three certificates in it
```
Comment (frankbu) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]:
```suggestion
    Open `httpbin-proxy-cert.txt`, which was created with the above command, and save the three certificates in it
```

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 10]
```diff
@@ -38,66 +35,53 @@ These files are ready to use in the `samples/certs/` directory.
...
-    In the following, we take the ratings pod as an example, and verify the certificates mounted on the pod.
-
-    Set the pod name to `RATINGSPOD`:
-
-    {{< text bash >}}
-    $ RATINGSPOD=`kubectl get pods -l app=ratings -o jsonpath='{.items[0].metadata.name}'`
-    {{< /text >}}
-
-    Run the following commands to retrieve the certificates mounted on the proxy:
+1. Deploy the `httpbin` and `sleep` example applications.
```
Comment (frankbu) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]:
```suggestion
1. Deploy the `httpbin` and `sleep` sample services.
```

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 11]
```diff
@@ -38,66 +35,53 @@ These files are ready to use in the `samples/certs/` directory.
...
-    $ istioctl manifest apply --set values.global.mtls.enabled=true --set values.security.selfSigned=false
-    {{< /text >}}
-
-1.  To make sure the workloads obtain the new certificates promptly,
-    delete the secrets generated by Citadel (named as `istio.\*`).
-    In this example, `istio.default`. Citadel will issue new certificates for the workloads.
-
-    {{< text bash >}}
-    $ kubectl delete secret istio.default
+    $ istioctl manifest apply --set profile=demo --set values.global.mtls.enabled=true
```
Comment (diemtvu) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Same as other, remove `--set values.global.mtls.enabled=true`

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md [hunk 12]
```diff
@@ -38,66 +35,53 @@ These files are ready to use in the `samples/certs/` directory.
...
     {{< /text >}}
 
-## Verifying the new certificates
+## Verifying the certificates
 
-In this section, we verify that the new workload certificates and root certificates are propagated.
+In this section, we verify that the workload certificates signed by the root certificates plugged in.
```
Comment (frankbu) [inline content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]:
```suggestion
In this section, we verify that workload certificates are signed by the certificates that we plugged into the CA.
```

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 4 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/ca-config/_index.md (hunk 5 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/_index.md]: Done.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 6 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 7 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 8 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 9 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 10 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 11 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: This guide requires mTLS (e.g., mTLS between sleep and httpbin sample services). If removing "--set values.global.mtls.enabled=true", how to make sure that mTLS is enforced?

File: content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md (hunk 12 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/ca-config/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
+---
+title: Plugging in External CA Key and Certificate
+description: Shows how operators can configure Istio's CA with existing root certificate, signing certificate and key.
+weight: 10
```
Comment (frankbu) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
weight: 80
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
+---
+title: Plugging in External CA Key and Certificate
+description: Shows how operators can configure Istio's CA with existing root certificate, signing certificate and key.
```
Comment (frankbu) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
description: Shows how operators can configure Istio's CA with an existing root certificate, signing certificate and key.
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
+---
+title: Plugging in External CA Key and Certificate
```
Comment (frankbu) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]: Configuring CA Certificates

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+description: Shows how operators can configure Istio's CA with existing root certificate, signing certificate and key.
+weight: 10
+keywords: [security,certificates]
+aliases:
+    - /docs/tasks/security/plugin-ca-cert/
+---
+
+This task shows how operators can configure the Istio certificate authority with an existing root certificate, signing certificate and key.
+
+By default, Istio's CA generates self-signed root certificate and key, and uses them to sign the workload certificates.
```
Comment (frankbu) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
By default, Istio's CA generates a self-signed root certificate and key, and uses them to sign the workload certificates.
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+Istio's CA can also use the operator-specified certificate and key to sign workload certificates, with
```
Comment (frankbu) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
Istio's CA can also sign workload certificates using an operator-specified certificate and key, and with an
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+operator-specified root certificate. This task demonstrates an example to plug certificates and key into Istio's CA.
```
Comment (frankbu) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
operator-specified root certificate. This task demonstrates how to plug such certificates and key into Istio's CA.
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+
+## Plugging in an existing certificate and key
+
+Suppose we want to have Istio's CA use an existing signing (CA) certificate `ca-cert.pem` and key `ca-key.pem`.
+Furthermore, the certificate `ca-cert.pem` is signed by the root certificate `root-cert.pem`.
+We would like to use `root-cert.pem` as the root certificate for Istio workloads.
+
+In the following example,
+Istio CA's signing (CA) certificate (`ca-cert.pem`) is different from root certificate (`root-cert.pem`),
```
Comment (frankbu) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
Istio CA's signing (CA) certificate (`ca-cert.pem`) is different from the root certificate (`root-cert.pem`),
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 13 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 14 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 15 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 16 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 17 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 18 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 19 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 20]
```diff
@@ -0,0 +1,117 @@
+---
+title: Configuring CA Certificates
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
title: Plugging in existing CA Certificates
```

Comment (myidpt) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: "Configuring" is very unclear. It could mean "configuring a self-signed cert" which is definitely not what you trying to say.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+description: Shows how operators can configure Istio's CA with an existing root certificate, signing certificate and key.
+weight: 80
+keywords: [security,certificates]
+aliases:
+    - /docs/tasks/security/plugin-ca-cert/
+---
+
+This task shows how operators can configure the Istio certificate authority with an existing root certificate, signing certificate and key.
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
This task shows how administrators can configure the Istio certificate authority with an existing root certificate, signing certificate and key.
```

Comment (myidpt) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: "Operator" could mean something different in Istio. Better to replace them with "administrator".

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 22]
```diff
@@ -0,0 +1,117 @@
+---
+title: Configuring CA Certificates
+description: Shows how operators can configure Istio's CA with an existing root certificate, signing certificate and key.
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
description: Shows how system administrators can configure Istio's CA with an existing root certificate, signing certificate and key.
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 23]
```diff
@@ -0,0 +1,117 @@
...
+weight: 80
+keywords: [security,certificates]
+aliases:
+    - /docs/tasks/security/plugin-ca-cert/
+---
+
+This task shows how operators can configure the Istio certificate authority with an existing root certificate, signing certificate and key.
+
+By default, Istio's CA generates a self-signed root certificate and key, and uses them to sign the workload certificates.
+Istio's CA can also sign workload certificates using an operator-specified certificate and key, and with an
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
Istio's CA can also sign workload certificates using an administrator-specified certificate and key, and with an
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 24]
```diff
@@ -0,0 +1,117 @@
...
+
+    {{< text bash >}}
+    $ kubectl exec $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name}) -c istio-proxy -n foo -- openssl s_client -showcerts -connect httpbin.foo:8000 > httpbin-proxy-cert.txt
+    {{< /text >}}
+
+    Open `httpbin-proxy-cert.txt`, which was created with the above command, and save the three certificates in it
+    to `proxy-cert-0.pem`, `proxy-cert-1.pem`, and `proxy-cert-2.pem`, respectively.
+    A certificate starts with `-----BEGIN CERTIFICATE-----` and ends with `-----END CERTIFICATE-----`.
+
+1.  Verify the root certificate is the same as the one specified by operator:
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
1.  Verify the root certificate is the same as the one specified by the administrator:
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+
+    {{< text bash >}}
+    $ openssl x509 -in @samples/certs/root-cert.pem@ -text -noout > /tmp/root-cert.crt.txt
+    $ openssl x509 -in ./proxy-cert-2.pem -text -noout > /tmp/pod-root-cert.crt.txt
+    $ diff /tmp/root-cert.crt.txt /tmp/pod-root-cert.crt.txt
+    {{< /text >}}
+
+    Expect the output to be empty.
+
+1.  Verify the CA certificate is the same as the one specified by operator:
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
1.  Verify the CA certificate is the same as the one specified by the administrator:
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 26]
```diff
@@ -0,0 +1,117 @@
...
+keywords: [security,certificates]
+aliases:
+    - /docs/tasks/security/plugin-ca-cert/
+---
+
+This task shows how operators can configure the Istio certificate authority with an existing root certificate, signing certificate and key.
+
+By default, Istio's CA generates a self-signed root certificate and key, and uses them to sign the workload certificates.
+Istio's CA can also sign workload certificates using an operator-specified certificate and key, and with an
+operator-specified root certificate. This task demonstrates how to plug such certificates and key into Istio's CA.
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
administrator-specified root certificate. This task demonstrates how to plug such certificates and key into Istio's CA.
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 27]
```diff
@@ -0,0 +1,117 @@
...
+
+## Plugging in an existing certificate and key
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
## Plugging in existing certificates and key
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 20 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 21 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 22 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 23 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 24 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 25 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 26 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 27 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 28]
```diff
@@ -0,0 +1,117 @@
...
+    {{< /text >}}
+
+    Open `httpbin-proxy-cert.txt`, which was created with the above command, and save the three certificates in it
+    to `proxy-cert-0.pem`, `proxy-cert-1.pem`, and `proxy-cert-2.pem`, respectively.
+    A certificate starts with `-----BEGIN CERTIFICATE-----` and ends with `-----END CERTIFICATE-----`.
+
+1.  Verify the root certificate is the same as the one specified by the administrator:
+
+    {{< text bash >}}
+    $ openssl x509 -in @samples/certs/root-cert.pem@ -text -noout > /tmp/root-cert.crt.txt
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]: What is "@" for?

Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: It will create a URL link to https://raw.githubusercontent.com/istio/istio/release-1.5/samples/certs/root-cert.pem.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 29]
```diff
@@ -0,0 +1,117 @@
...
+    $ openssl x509 -in ./proxy-cert-1.pem -text -noout > /tmp/pod-cert-chain-ca.crt.txt
+    $ diff /tmp/ca-cert.crt.txt /tmp/pod-cert-chain-ca.crt.txt
+    {{< /text >}}
+
+    Expect the output to be empty.
+
+1.  Verify the certificate chain from the root certificate to the workload certificate:
+
+    {{< text bash >}}
+    $ head -n 21 ./proxy-cert-0.pem > /tmp/pod-cert-chain-workload.pem
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]: Don't need `head` here. proxy-cert-0.pem has a single cert already.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 28 above)
Comment (myidpt) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: I see. Thanks.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md [hunk 30]
```diff
@@ -0,0 +1,117 @@
...
+    $ openssl verify -CAfile <(cat @samples/certs/ca-cert.pem@ @samples/certs/root-cert.pem@) /tmp/pod-cert-chain-workload.pem
```
Comment (myidpt) [inline content/en/docs/tasks/security/plugin-ca-cert/index.md]:
```suggestion
    $ openssl verify -CAfile <(cat @samples/certs/ca-cert.pem@ @samples/certs/root-cert.pem@) ./proxy-cert-0.pem
```

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 29 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.

File: content/en/docs/tasks/security/plugin-ca-cert/index.md (hunk 30 above)
Comment (lei-tang) [reply content/en/docs/tasks/security/plugin-ca-cert/index.md]: Done.
